### PR TITLE
ColorUtil Enhancements

### PR DESCRIPTION
**Pull request by oetiker**
* handle hex4 and hex8 wherever hex3 and hex6 have been expected until now
* remove alpha if it is 1
* new functions `scale` and `adjust` modeled after the same functions in sass. They let you modify individual properties of an incoming color.

### CODE REVIEW

**Review by cboulanger — APPROVED** [review]

**Review by hkollmann — APPROVED** [review]

**Comment by hkollmann** [comment]
Needs some linter fixes....

File: framework/source/class/qx/util/ColorUtil.js [hunk 1]
```diff
@@ -164,83 +166,63 @@ qx.Bootstrap.define("qx.util.ColorUtil",
...
+      var color;
       if (this.isNamedColor(str))
       {
-        return this.NAMED[str];
+        color = this.NAMED[str].concat();
       }
       else if (this.isSystemColor(str))
       {
-        throw new Error("Could not convert system colors to RGB: " + str);
+          throw new Error("Could not convert system colors to RGB: " + str);
```
Comment (derrell) [inline framework/source/class/qx/util/ColorUtil.js:202]: Why is it an error if it's a system color?

**Review by derrell — CHANGES_REQUESTED** [review]
This looks like it will be a nice enhancement. I propose, however, that we treat the current master branch as feature-locked except for bug fixes (and the occasional late non-BC change) until release. Can this hold off for a 6.1 release?

File: framework/source/class/qx/util/ColorUtil.js (hunk 1 above)
Comment (oetiker) [reply framework/source/class/qx/util/ColorUtil.js:202]: I have no idea ... that has been there before

**Comment by oetiker** [comment]
> This looks like it will be a nice enhancement. I propose, however, that we treat the current master branch as feature-locked except for bug fixes (and the occasional late non-BC change) until release. Can this hold off for a 6.1 release?

we can, but then we also have to hold off on the new theme ...

**Comment by cboulanger** [comment]
@derrell You're right in principle, but I think we should have a new theme for the release and this appears necessary for the theme to work...

**Comment by derrell** [comment]
I agree. The new theme has been in the planning for a long time, and I didn't realize that this was part of that work.

**Review by derrell — APPROVED** [review]

**Review by zaucker — APPROVED** [review]

**Review by johnspackman — APPROVED** [review]

**Pull request merged**